plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -308,8 +332,20 @@ public CompletableFuture<Collection<Slice>> finish()
                 .map(Slices::wrappedBuffer)
                 .forEach(fragments::add);
 
-        fileDeletions.forEach((path, deletion) ->
-                fragments.addAll(rewriteFile(path.toStringUtf8(), deletion)));
+        fileDeletions.forEach((path, deletion) -> {
+            if (deletionVectorEnabled) {
+                Roaring64Bitmap pastDeletionVectors = loadDeletionVectors(Location.of(path.toStringUtf8()));
+                if (isNoRowsRetained(path.toStringUtf8(), pastDeletionVectors, deletion)) {
+                    fragments.add(onlySourceFile(path.toStringUtf8(), deletion));
+                }
+                else {
+                    fragments.add(writeDeletionVector(path.toStringUtf8(), pastDeletionVectors, deletion));
```

Comment:
I would suggest combining this if/else into a single code path.
Maybe at a high level it could look like
```
OR the deleted row bitmaps;
Get file row count from footer;
deletionVectorForWrite = filePositions.andNot(deletedRows);
if deletionVectorForWrite is empty, no need to write it, otherwise write this vector;
```
This way we won't repeat the read of parquet footer and OR'ing of bitmaps in both code paths